File: tests/test_files/java/WeakCryptoComprehensive.java
package com.example.weakcrypto;

import javax.crypto.Cipher;
import javax.crypto.KeyGenerator;
import javax.crypto.SecretKey;
import javax.crypto.spec.SecretKeySpec;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.security.SecureRandom;
import java.util.Random;

public class WeakCryptoComprehensive {
    
    // ==================== Weak Hash Algorithms ====================
    public void weakHashAlgorithms() throws NoSuchAlgorithmException {
        // These should trigger weak crypto rules
        MessageDigest md5 = MessageDigest.getInstance("MD5");
        MessageDigest sha1 = MessageDigest.getInstance("SHA-1");
        MessageDigest sha1Alt = MessageDigest.getInstance("SHA1");
        
        // These should be fine
        MessageDigest sha256 = MessageDigest.getInstance("SHA-256");
        MessageDigest sha512 = MessageDigest.getInstance("SHA-512");
    }
    
    // ==================== Weak Encryption Algorithms ====================
    public void weakEncryptionAlgorithms() throws NoSuchAlgorithmException {
        // These should trigger weak crypto rules
        Cipher desCipher = Cipher.getInstance("DES");
        Cipher rc4Cipher = Cipher.getInstance("RC4");
        Cipher arc4Cipher = Cipher.getInstance("ARC4");
        Cipher tripleDesCipher = Cipher.getInstance("DESede");
        Cipher tripleDesAlt = Cipher.getInstance("3DES");
        
        // These should be fine
        Cipher aesCipher = Cipher.getInstance("AES/GCM/NoPadding");
        Cipher aesCbcCipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
    }
    
    // ==================== Weak Encryption Modes ====================
    public void weakEncryptionModes() throws NoSuchAlgorithmException {
        // These should trigger weak crypto rules
        Cipher ecbAes = Cipher.getInstance("AES/ECB/PKCS5Padding");
        Cipher ecbDes = Cipher.getInstance("DES/ECB/PKCS5Padding");
        Cipher ecb3Des = Cipher.getInstance("DESede/ECB/PKCS5Padding");
        
        // These should be fine
        Cipher cbcAes = Cipher.getInstance("AES/CBC/PKCS5Padding");
        Cipher gcmAes = Cipher.getInstance("AES/GCM/NoPadding");
    }
    
    // ==================== Weak Random Number Generation ====================
    public void weakRandomGeneration() {
        // These should trigger weak crypto rules
        Random weakRandom = new Random();
        int weakInt = weakRandom.nextInt();
        long weakLong = weakRandom.nextLong();
        double weakDouble = weakRandom.nextDouble();
        
        // These should be fine
        SecureRandom secureRandom = new SecureRandom();
        int secureInt = secureRandom.nextInt();
        byte[] secureBytes = new byte[16];
        secureRandom.nextBytes(secureBytes);
    }
    
    // ==================== Hardcoded Cryptographic Material ====================
    public void hardcodedSecrets() throws NoSuchAlgorithmException {
        // These should trigger hardcoded key rules
        String key = "my-secret-key-12345";
        String password = "admin123";
        String secret = "super-secret-token";
        String token = "jwt-token-here";
        String privateKey = "-----BEGIN PRIVATE KEY-----";
        String publicKey = "-----BEGIN PUBLIC KEY-----";
        
        // Use the hardcoded values in cryptographic operations
        SecretKeySpec secretKey = new SecretKeySpec(key.getBytes(), "AES");
        MessageDigest md = MessageDigest.getInstance("SHA-256");
        md.update(password.getBytes());
    }
    
    // ==================== Weak Key Sizes ====================
    public void weakKeySizes() throws NoSuchAlgorithmException {
        // These should trigger weak key size rules
        KeyGenerator rsaGen = KeyGenerator.getInstance("RSA");
        rsaGen.init(1024); // Weak RSA key size
        
        // This should be fine
        KeyGenerator aesGen = KeyGenerator.getInstance("AES");
        aesGen.init(256); // Strong AES key size
    }
    
    // ==================== Taint Flow Examples ====================
    public void taintFlowExamples(String userInput) throws NoSuchAlgorithmException {
        // These should trigger taint flow rules
        String hardcodedKey = "hardcoded-secret-key";
        SecretKeySpec key = new SecretKeySpec(hardcodedKey.getBytes(), "AES");
        Cipher cipher = Cipher.getInstance("AES");
        
        // User input flowing to cryptographic operations
        MessageDigest md = MessageDigest.getInstance("SHA-256");
        md.update(userInput.getBytes());
    }
    
    // ==================== Safe Examples (should not trigger rules) ====================
    public void safeCryptographicUsage() throws NoSuchAlgorithmException {
        // Strong hash algorithms
        MessageDigest sha256 = MessageDigest.getInstance("SHA-256");
        MessageDigest sha512 = MessageDigest.getInstance("SHA-512");
        
        // Strong encryption algorithms
        Cipher aesCipher = Cipher.getInstance("AES/GCM/NoPadding");
        Cipher aesCbcCipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
        
        // Secure random number generation
        SecureRandom secureRandom = new SecureRandom();
        byte[] randomBytes = new byte[32];
        secureRandom.nextBytes(randomBytes);
        
        // Strong key sizes
        KeyGenerator aesGen = KeyGenerator.getInstance("AES");
        aesGen.init(256);
    }
}

File: tests/test_files/javascript/WeakCryptoComprehensive.js
// ==================== Weak Hash Algorithms ====================
function weakHashAlgorithms() {
    const crypto = require('crypto');
    const CryptoJS = require('crypto-js');
    
    // These should trigger weak crypto rules
    const md5Hash = crypto.createHash('md5');
    const md5HashAlt = crypto.createHash('MD5');
    const sha1Hash = crypto.createHash('sha1');
    const sha1HashAlt = crypto.createHash('SHA1');
    
    // CryptoJS weak algorithms
    const md5CryptoJS = CryptoJS.MD5('test');
    const sha1CryptoJS = CryptoJS.SHA1('test');
    
    // These should be fine
    const sha256Hash = crypto.createHash('sha256');
    const sha512Hash = crypto.createHash('sha512');
    const sha256CryptoJS = CryptoJS.SHA256('test');
}

// ==================== Weak Encryption Algorithms ====================
function weakEncryptionAlgorithms() {
    const crypto = require('crypto');
    const CryptoJS = require('crypto-js');
    
    // These should trigger weak crypto rules
    const desCipher = crypto.createCipher('des', 'password');
    const desDecipher = crypto.createDecipher('des', 'password');
    const rc4Cipher = crypto.createCipher('rc4', 'password');
    const arc4Cipher = crypto.createCipher('arc4', 'password');
    const tripleDesCipher = crypto.createCipher('des3', 'password');
    
    // CryptoJS weak algorithms
    const desCryptoJS = CryptoJS.DES.encrypt('message', 'password');
    const rc4CryptoJS = CryptoJS.RC4.encrypt('message', 'password');
    const tripleDesCryptoJS = CryptoJS.TripleDES.encrypt('message', 'password');
    
    // These should be fine
    const aesCipher = crypto.createCipher('aes-256-cbc', 'password');
    const aesGcmCipher = crypto.createCipher('aes-256-gcm', 'password');
    const aesCryptoJS = CryptoJS.AES.encrypt('message', 'password');
}

// ==================== Weak Encryption Modes ====================
function weakEncryptionModes() {
    const crypto = require('crypto');
    const CryptoJS = require('crypto-js');
    
    // These should trigger weak crypto rules
    const ecbAes = crypto.createCipher('aes-256-ecb', 'password');
    const ecbDes = crypto.createCipher('des-ecb', 'password');
    const ecb3Des = crypto.createCipher('des3-ecb', 'password');
    
    // CryptoJS ECB mode
    const ecbCryptoJS = CryptoJS.AES.encrypt('message', 'password', { mode: CryptoJS.mode.ECB });
    
    // These should be fine
    const cbcAes = crypto.createCipher('aes-256-cbc', 'password');
    const gcmAes = crypto.createCipher('aes-256-gcm', 'password');
    const cbcCryptoJS = CryptoJS.AES.encrypt('message', 'password', { mode: CryptoJS.mode.CBC });
}

// ==================== Weak Random Number Generation ====================
function weakRandomGeneration() {
    // These should trigger weak crypto rules
    const weakRandom = Math.random();
    const weakInt = Math.floor(Math.random() * 100);
    const weakCeil = Math.ceil(Math.random() * 100);
    const weakRound = Math.round(Math.random() * 100);
    const timeBased = Date.now();
    const performanceBased = performance.now();
    
    // These should be fine
    const secureRandom = crypto.getRandomValues(new Uint8Array(16));
    const secureRandom32 = crypto.getRandomValues(new Uint32Array(4));
}

// ==================== Hardcoded Cryptographic Material ====================
function hardcodedSecrets() {
    // These should trigger hardcoded key rules
    const key = "my-secret-key-12345";
    const password = "admin123";
    const secret = "super-secret-token";
    const token = "jwt-token-here";
    const apiKey = "api-key-12345";
    const privateKey = "-----BEGIN PRIVATE KEY-----";
    const publicKey = "-----BEGIN PUBLIC KEY-----";
    
    // Use the hardcoded values in cryptographic operations
    const crypto = require('crypto');
    const hash = crypto.createHash('sha256');
    hash.update(key);
    
    const cipher = crypto.createCipher('aes-256-cbc', password);
}

// ==================== Weak Key Derivation ====================
function weakKeyDerivation() {
    const crypto = require('crypto');
    const CryptoJS = require('crypto-js');
    
    // These should trigger weak key derivation rules
    const md5Hash = crypto.createHash('md5').update('password').digest();
    const sha1Hash = crypto.createHash('sha1').update('password').digest();
    const md5CryptoJS = CryptoJS.MD5('password');
    const sha1CryptoJS = CryptoJS.SHA1('password');
    
    // These should be fine
    const pbkdf2 = crypto.pbkdf2Sync('password', 'salt', 100000, 64, 'sha512');
    const scrypt = crypto.scryptSync('password', 'salt', 64);
}

// ==================== Insecure JWT Usage ====================
function insecureJWTUsage() {
    const jwt = require('jsonwebtoken');
    
    // These should trigger insecure JWT rules
    const token1 = jwt.sign({ user: 'test' }, 'secret', { algorithm: 'none' });
    const token2 = jwt.sign({ user: 'test' }, 'secret', { algorithm: 'HS256' });
    const token3 = jwt.sign({ user: 'test' }, 'secret', { algorithm: 'HS384' });
    const token4 = jwt.sign({ user: 'test' }, 'secret', { algorithm: 'HS512' });
    
    // These should be fine
    const secureToken = jwt.sign({ user: 'test' }, 'secret', { algorithm: 'RS256' });
    const secureToken2 = jwt.sign({ user: 'test' }, 'secret', { algorithm: 'ES256' });
}

// ==================== Taint Flow Examples ====================
function taintFlowExamples(userInput) {
    const crypto = require('crypto');
    
    // These should trigger taint flow rules
    const hardcodedKey = "hardcoded-secret-key";
    const hash = crypto.createHash('sha256');
    hash.update(hardcodedKey);
    
    const cipher = crypto.createCipher('aes-256-cbc', hardcodedKey);
    
    // User input flowing to cryptographic operations
    const userHash = crypto.createHash('sha256');
    userHash.update(userInput);
}

// ==================== Safe Examples (should not trigger rules) ====================
function safeCryptographicUsage() {
    const crypto = require('crypto');
    const CryptoJS = require('crypto-js');
    
    // Strong hash algorithms
    const sha256Hash = crypto.createHash('sha256');
    const sha512Hash = crypto.createHash('sha512');
    const sha256CryptoJS = CryptoJS.SHA256('test');
    
    // Strong encryption algorithms
    const aesCipher = crypto.createCipher('aes-256-cbc', 'password');
    const aesGcmCipher = crypto.createCipher('aes-256-gcm', 'password');
    const aesCryptoJS = CryptoJS.AES.encrypt('message', 'password');
    
    // Secure random number generation
    const secureRandom = crypto.getRandomValues(new Uint8Array(32));
    
    // Strong key derivation
    const pbkdf2 = crypto.pbkdf2Sync('password', 'salt', 100000, 64, 'sha512');
    const scrypt = crypto.scryptSync('password', 'salt', 64);
    
    // Secure JWT
    const jwt = require('jsonwebtoken');
    const secureToken = jwt.sign({ user: 'test' }, 'secret', { algorithm: 'RS256' });
}
